dev/RELEASE_PROCESS.rst

```diff
@@ -3,25 +3,65 @@ Release Process
 
 This document describes the process for creating new releases.
 
-1. First, you should build wheels that you'd like to use for testing. That can
-   be done by following the `documentation for building wheels`_.
+1. **Increment the Python version:** Create a PR that increments the Python
+   package version. See `this example`_.
+
+2. **Download the Travis-built wheels:** Once The wheels from this commit can be
```

Comment:
`Once The wheels...` there is something missing in this sentence :)

Reply:
Agreed, this paragraph needs rephrasing.